sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorReceiver.java

```diff
@@ -147,75 +215,86 @@ public boolean isDisposed() {
 
     @Override
     public void dispose() {
-        if (isDisposed.getAndSet(true)) {
-            return;
-        }
+        dispose("Dispose invoked", null)
+            .block(retryOptions.getTryTimeout());
```

Comment:
This `.block` call may cause exception when `dispose` is called in async code. For instance, `AmqpReceiveLinkProcessor` calls it and the following exception was thrown. There are other places use .block() too. I guess they all need to be changed.

java.lang.IllegalStateException: block()/blockFirst()/blockLast() are blocking, which is not supported in thread reactor

Reply:
This should not happen as this dispose method is a synchronous call. If this causes an issue with blocking then this method should return `Mono<Void>`.